docs/book/config.md

```diff
@@ -26,10 +26,10 @@ return [
         // as specified above.
         'routes' => [
             // Production path.
-            // 'production' => '/{provider:facebook|github|google|instagram}|linkedin[/oauth2callback]',
+            // 'production' => '/{provider:facebook|github|google|instagram|linkedin|custom}[/oauth2callback]',
 
             // Debug path.
-            // 'debug' => '/{provider:debug|facebook|github|google|instagram|linkedin}[/oauth2callback]',
+            // 'debug' => '/{provider:debug|facebook|github|google|instagram|linkedin|custom}[/oauth2callback]',
```

Comment:
You should likely add a note that the string "custom" is not what will be matched, and that what should be in this regex are the _keys_ associated with providers in configuration.

Reply:
Ok, something like?: 
```php
// Provider names in the regex should match the keys associated with enabled providers in configuration. The "custom" string is just an example of a provider key named "custom" described below.
```

Note: The string "custom" would actually be matched if one followed the example of how to setup a GenericProvider further down in the [Local/Environment-specific configuration](https://github.com/nynka/phly-expressive-oauth2clientauthentication/blob/support-upstream-clients/docs/book/config.md#localenvironment-specific-configuration) section.

```php
// Customized
// 'custom' => [
//    'provider' => Provider\GenericProvider::class,
//    'options' => [
//        'clientId' => '',
//        'clientSecret' => '',
//        'redirectUri' => '',
//        'urlAuthorize' => '',
//        'urlAccessToken' => '',
//        'urlResourceOwnerDetails' => '',
//    ],
// ],
```

Reply:
Yes, perfect!